### ISSUE TITLE
Regression: Type Inference fails again on nested zip calls

### ISSUE THREAD

**Issue by pvandyken** [microsoft/pyright#5947]
Apologies for making what looks like a duplicate bug report: in #5780 I reported type inference was failing in nested zip calls. The issue was fixed in `pyright==1.1.324`, but it reappeared in `pyright==1.1.325` and is still present in v1.1.327.

For reference, here's the failing example again:

(note: bar is still getting typed as `zip[tuple[str]]`, where I believe it should be `zip[tuple[str, ...]]`. I'm not sure if this was fixed in 1.1.324 or not).

Version: `>=1.1.325,<=1.1.327`
```py
# test.py
from __future__ import annotations
from typing import Iterable


def foo(a: Iterable[tuple[str, ...]]):
    bar = zip(*a)
    # zip[tuple[str]]

    _result = zip(a, bar)
    # zip[tuple[tuple[str, ...], tuple[str]]]

    zip(a, zip(*a))
	#      ~~~~~~~
    # Argument type is partially unknown
    #   Argument corresponds to parameter "__iter2" in function "__new__"
    #   Argument type is "zip[tuple[Unknown]]
```

Output:

```
/path/to/test.py
  /path/to/test.py:12:12 - error: Argument type is partially unknown
  Argument corresponds to parameter "__iter2" in function "__new__"
  Argument type is "zip[tuple[Unknown]]
```

**Comment by erictraut** [comment]
Thanks for reporting, and apologies for the regression.

I've localized the problem to the commit that fixed this bug: https://github.com/microsoft/pyright/issues/5804.

**Comment by erictraut** [comment]
This will be addressed in the next release. I've added your code sample to the unit tests to catch and prevent similar regressions in the future.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.328, which I just published. It will also be included in a future release of pylance.
